### ISSUE TITLE
Add support for fractional and notional orders

### ISSUE THREAD

**Issue by ktrauberman** [alpacahq/alpaca-trade-api-csharp#294]
Now that [fractional and notional orders are supported](https://alpaca.markets/blog/fractional-trading/) the `PostOrderAsync` method should be updated to support them.

**Comment by OlegRa** [comment]
@ktrauberman I'm aware of these changes but unfortunately, it's not easy to support them in a backward-compatible way so most probably I'll add this support in 5.0 first, and later if I'll come with some idea how to handle it without breaking existing code I'll port back these changes into 4.1 and (maybe) in 4.0 too. But at least the deserialization code should be ready for non-integer quantities for orders so most probably I'll make a hotfix release of 4.x very soon.

**Comment by OlegRa** [comment]
@ktrauberman Please, test and close:

- Version [4.1.2](../releases/tag/sdk-4.1.2) contains full support for fractional orders including new classes/methods for specifying fractional/notional values on market order placement.
- Version [4.0.6](../releases/tag/sdk-4.0.6) contains only limited support for not crashing on fractional sizes deserialiyation.

**Comment by ktrauberman** [comment]
Thanks for the quick turnaround @OlegRa 

I was able to successfully place a fractional order using `await tradingClient.PostOrderAsync(new NewOrderRequest("TSLA", OrderQuantity.Fractional(0.1m), OrderSide.Buy, OrderType.Market, TimeInForce.Day));`

However, I got an error while placing a notional order using `var order = await tradingClient.PostOrderAsync(new NewOrderRequest("TSLA", OrderQuantity.Notional(100m), OrderSide.Buy, OrderType.Market, TimeInForce.Day));` 

The error I got was:
>Newtonsoft.Json.JsonSerializationException: Error converting value {null} to type 'System.Decimal'. Path 'qty', line 1, position 477.
 ---> System.InvalidCastException: Null object cannot be converted to a value type.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
   --- End of inner exception stack trace ---
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at Alpaca.Markets.HttpResponseMethodExtensions.DeserializeAsync[TApi,TJson](HttpResponseMessage response)
   at Alpaca.Markets.HttpResponseMethodExtensions.DeserializeAsync[TApi,TJson](HttpResponseMessage response)
   at Alpaca.Markets.HttpClientExtensions.callAndDeserializeAsync[TApi,TJson](HttpClient httpClient, HttpRequestMessage request, CancellationToken cancellationToken, IThrottler throttler)
   at Alpaca.Markets.HttpClientExtensions.callAndDeserializeAsync[TApi,TJson,TContent](HttpClient httpClient, HttpMethod method, Uri endpointUri, TContent content, CancellationToken cancellationToken, IThrottler throttle)

Note that the notional order was submitted successfully.  The error appears to simply be an error deserializing the json returned by the postOrder endpoint.
